builtin/aws/apprunner/platform.go

```diff
@@ -0,0 +1,445 @@
+	if port == 0 {
+		// App Runner default port
+		port = 8080
+	}
+
+	roleName := p.config.RoleName
+	if roleName == "" {
+		// if role name is not specified, we'll fall back to
+		// getting or creating a service role, with the same
+		// default roleName that AWS uses in the console
+		roleName = defaultRoleName
+	}
+
+	// roleArn only applies to the initial create
+	var roleArn string
+	if arn, err := p.getOrCreateIamRoleArn(sess, log, roleName); err != nil {
+		return nil, err
+	} else {
+		roleArn = arn
+	}
+
+	mem := int64(p.config.Memory)
+	if mem == 0 {
+		mem = defaultMemory
+	}
+
+	cpu := int64(p.config.Cpu)
+	if cpu == 0 {
+		cpu = defaultCpu
+	}
+
+	envVars := make(map[string]*string)
+	for k, v := range p.config.StaticEnvVars {
+		envVars[k] = aws.String(v)
+	}
+
+	step.Done()
+
+	step = sg.Add("Deploying App Runner service %q", p.config.Name)
+	step.Done()
+
+	arSvc := apprunner.New(sess)
+
+	// The operator's waypoint.hcl will realistically only have a svc name, not ARN,
+	// since that is provided by AWS.
+	//
+	// While `DescribeService` only supports a service ARN, our best effort approach
+	// is to list all services and manually match by user-provided name.
+	step = sg.Add("Checking for existing service...")
+	service, err := p.getServiceSummaryByName(sess, log, p.config.Name)
+	if err != nil {
+		return nil, err
+	}
+	step.Done()
+
+	operationId := ""
+	serviceArn := ""
+	serviceUrl := ""
+	serviceStatus := ""
+
+	// If we found a previous service, update it.
+	if service != nil {
+		step = sg.Add("Found! Updating service %q", service.Name)
+		serviceArn = service.Arn
+		serviceUrl = service.Url
+		serviceStatus = service.Status
+
+		uso, err := arSvc.UpdateService(&apprunner.UpdateServiceInput{
+			ServiceArn: aws.String(service.Arn),
+			InstanceConfiguration: &apprunner.InstanceConfiguration{
+				Cpu:    aws.String(strconv.FormatInt(cpu, 10)),
+				Memory: aws.String(strconv.FormatInt(mem, 10)),
+			},
+			SourceConfiguration: &apprunner.SourceConfiguration{
+				AuthenticationConfiguration: &apprunner.AuthenticationConfiguration{
+					AccessRoleArn: aws.String(roleArn),
+				},
+				ImageRepository: &apprunner.ImageRepository{
+					ImageRepositoryType: aws.String(apprunner.ImageRepositoryTypeEcr),
+					ImageIdentifier:     aws.String(img.Name()),
+					ImageConfiguration: &apprunner.ImageConfiguration{
+						Port:                        aws.String(strconv.Itoa(port)),
+						RuntimeEnvironmentVariables: envVars,
+					},
+				},
+				AutoDeploymentsEnabled: aws.Bool(false),
+			},
+		})
+
+		if err != nil {
+			step.Update("Failed to update service: %s", err)
+			if aerr, ok := err.(awserr.Error); ok {
+				if aerr.Code() == apprunner.ErrCodeInvalidStateException {
+					// If state is CREATE_FAILED, the service likely should be manually deleted,
+					// and then re-created.
+
+					// A substring check for "CREATE_FAILED" is our best effort to detect this
+					if strings.Contains(aerr.Message(), "CREATE_FAILED") {
```

Comment:
If it ends up in this state, could `waypoint destroy` handle this instead of manual deletion?

Reply:
@paladin-devops I'm not certain if `waypoint destroy` _can_ but maybe you can help confirm/deny? (Context below 👇 )

(I've since moved the `Destroy` functionality from release-manager to the platform — I don't think this changes too much.)

Thinking out load + I have loose understanding here, so take this with a grain of salt: The `Destroy` function depends on a `Deployment` protobuf so that it can access the `.ServiceArn` field to know which service to delete.
If `Deploy` fails,would `Destroy` have access to a `deployment *Deployment` protobuf? 
- I think this state is only expected to ever occur on a _very first_ `waypoint up` for an AppRunner project.